### PR TITLE
Refactor to add new Elasticsearch API Client

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/5210

### CODE REVIEW

File: crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/client/ElasticsearchClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/client/ElasticsearchClientWrapper.java:2]: Please update the copyright year.

**Review by avasquez614 — APPROVED** [review]

File: crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/impl/client/AbstractElasticsearchClientWrapper.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/impl/client/AbstractElasticsearchClientWrapper.java:2]: 2022

File: crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/spring/RestHighLevelClientFactory.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/spring/RestHighLevelClientFactory.java:2]: 2022

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please also fix SonaType Lift issues.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**